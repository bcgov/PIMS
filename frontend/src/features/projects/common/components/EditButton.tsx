import * as React from 'react';
import { Button } from 'react-bootstrap';

interface IEditButtonProps {
  /** whether or not the form this button belongs to is disabled */
  formDisabled?: boolean;
  /** set the form to be editable */
  setFormDisabled?: Function;
  /** true if the current user is allowed to edit this form */
  canEdit?: boolean;
}

/**
 * Button that allows toggling the form state (editable/not-editable)
 * @param param0 IEditButtonPropts
 */
const EditButton: React.FunctionComponent<IEditButtonProps> = ({
  formDisabled,
  setFormDisabled,
}: IEditButtonProps) => {
  return setFormDisabled ? (
    <Button disabled={!formDisabled} className="edit" onClick={() => setFormDisabled(false)}>
      Edit
    </Button>
  ) : null;
};

export default EditButton;
