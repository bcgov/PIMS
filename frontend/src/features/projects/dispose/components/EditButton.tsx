import * as React from 'react';
import { Button } from 'react-bootstrap';

interface IEditButtonProps {
  /** whether or not the form this button belongs to is disabled */
  formDisabled?: boolean;
  /** set the form to be editable */
  setFormDisabled: Function;
  /** true if the current user is allowed to edit this form */
  canEdit?: boolean;
}

/**
 * Button that allows toggling the form state (editable/not-editable)
 * @param param0 IEditButtonPropts
 */
const EditButton: React.FunctionComponent<IEditButtonProps> = ({
  formDisabled: disabled,
  setFormDisabled: setDisabled,
  canEdit,
}: IEditButtonProps) => {
  return canEdit ? (
    <Button disabled={!disabled} className="edit" onClick={() => setDisabled(false)}>
      Edit
    </Button>
  ) : null;
};

export default EditButton;
