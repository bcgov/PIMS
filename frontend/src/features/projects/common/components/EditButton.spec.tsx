import { cleanup, render } from '@testing-library/react';
import { noop } from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import EditButton from './EditButton';

const getEditButton = (setFormDisabled?: Function, formDisabled?: boolean) => {
  return <EditButton {...{ formDisabled, setFormDisabled }} />;
};

describe('Project Dispose Form Edit Button', () => {
  afterEach(() => {
    cleanup();
  });
  it('Matches Snapshot', () => {
    const component = renderer.create(getEditButton(noop, true));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('The button is disabled if the form is not disabled', () => {
    const { getByText } = render(getEditButton(noop, true));
    const editButton = getByText('Edit');
    expect(editButton).toBeTruthy();
    expect(editButton).toBeVisible();
  });

  it('Edit Button is not visible if setIsSubmitting is undefined', () => {
    const { queryByText } = render(getEditButton(undefined, false));
    const editButton = queryByText('Edit');
    expect(editButton).toBeNull();
  });
});
