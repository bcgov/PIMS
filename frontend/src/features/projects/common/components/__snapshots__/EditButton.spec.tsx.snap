// Vitest Snapshot v1
<<<<<<<< HEAD:frontend/src/features/projects/common/components/__snapshots__/EditButton.spec.tsx.snap
========

exports[`Project Dispose Form Edit Button > Matches Snapshot 1`] = `
<button
  className="edit btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Edit
</button>
`;
>>>>>>>> ef22e273 (Changed jest to vitest, 17 tests failing):frontend/src/features/projects/common/components/__snapshots__/EditButton.test.tsx.snap

exports[`Project Dispose Form Edit Button > Matches Snapshot 1`] = `
<button
  className="edit btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Edit
</button>
`;
